### ISSUE TITLE
Optional parameter with a message attribute exception

### ISSUE THREAD

**Issue by ericproulx** [ruby-grape/grape#2517]
It probably never happened but defining a optional parameter with a message attribute will raise an exception.

```ruby
context 'optional param with message' do
  let(:app) do
    Class.new(described_class) do
      params do
        optional 'test', message: 'is required'
      end
      get 'test'
    end
  end

  before do

  end

  it 'should return a message' do
    expect { get 'test' }.not_to raise_error
  end
end
```

**Comment by wwqwq2313** [comment]
Hello, we tried to solve the issue.

This is what we did:

Modify the `optional` method in `Grape::DSL::Parameters` to correctly handle the `message` option for optional parameters without raising an exception.

You can review changes in this commit: https://github.com/wwqwq2313/ruby-grape-grape-2517/commit/6e1e8f88042b9ee36e94ee523a7022a73584e63e.


> [!CAUTION]
> Disclaimer: The concept of solution was created by AI and you should never copy paste this code before you check the correctness of generated code. Solution might not be complete, you should use this code as an inspiration only.

---

Latta AI seeks to solve problems in open source projects as part of its mission to support developers around the world. Learn more about our mission at https://latta.ai/ourmission . If you no longer want Latta AI to attempt solving issues on your repository, you can block this account.

**Comment by ericproulx** [comment]
@wwqwq2313 almost, but since :message act only on required field, we just need to delete it on optional field. Also, when adding :presence, we should delete the `:message` the validations.

**Comment by dblock** [comment]
I think @wwqwq2313 is an AI (!)

**Comment by ericproulx** [comment]
@dblock yeah I know. I just wanted to know if it would respond back after my comment.
